addons/auth_totp_mail_enforce/tests/test_auth_signup.py

```diff
@@ -0,0 +1,40 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.addons.base.tests.common import HttpCaseWithUserDemo, HttpCaseWithUserPortal
+
+
+class TestAuthSignupFlowWith2faEnforced(HttpCaseWithUserPortal, HttpCaseWithUserDemo):
+
+    def setUp(self):
+        super().setUp()
+        res_config = self.env['res.config.settings']
+        self.default_values = res_config.default_get(list(res_config.fields_get()))
+
+        # Activate free signup
+        self.default_values.update({'auth_signup_uninvited': 'b2c'})
+        # Enforce 2FA for all users
+        self.default_values.update({'auth_totp.policy': 'all_required'})
```

Comment:
Still something fishy with this test... Now it seems to pass with or without your fix: I take your branch, I remove the fix but keep the test and do:
`-i auth_totp_mail_enforce --test-tags=.test_signup_with_2fa_enforced` -> No issue.

![image](https://github.com/odoo/odoo/assets/10338094/50f1c650-eb7e-4578-b0ff-03273e39ee4a)

But I don't know why :grimacing: I wonder if your setup is done properly, I don't understand how it works: you update a `default_values` object but I don't get where it is used in the codebase. Is the system parameter properly set doing that ? :thinking:

Reply:
It should be fixed now (I tested it thoroughly) - along with being able to run the test as many times as you want without resetting the DB.

It previously passed without the fix because the config parameters weren't the right ones (difference between `res.config.settings` and `ir.config_parameter`).

I don't remember why I removed the `@tagged` `post_install` but not having it is what made it fail with some error about not having access to some assets.